src/lib/db/group-store.ts

```diff
@@ -53,13 +55,15 @@ const rowToGroupUser = (row) => {
         groupId: row.group_id,
         joinedAt: row.created_at,
         createdBy: row.created_by,
+        rootRoleId: row.root_role_id,
     };
 };
 
 const groupToRow = (group: IStoreGroup) => ({
     name: group.name,
     description: group.description,
     mappings_sso: JSON.stringify(group.mappingsSSO),
+    root_role_id: group.rootRole || null,
```

Comment:
This is a problem with our validation layer. We can't send `null` numbers without them being coerced to `0`. So we don't send the property from the front end and if we don't find it, it gets coerced to `null` so that when we update the group the root role can be removed. In practise this is only done through a `PUT` endpoint so it feels okay. Happy to be told I'm mad

Reply:
Yep, reminds me of discussions we had like https://unleash-internal.slack.com/archives/C048ELND3QD/p1678447540849379 - May be worth having a look at how we're handling `project` in segments to see if we can keep it consistent.

Reply:
Yeah, I remember you guys talking about this one. It's a very similar approach - I had a chat with Gaston before I took a stab at this. The subtle differences are mostly due string vs number things in the validation layer